File: easy/problems/0349-intersection-of-two-arrays/solution.ts
function intersection(nums1: number[], nums2: number[]): number[] {
    const set1 = new Set(nums1);
    const result: number[] = [];

    for (const num of nums2) {
        if (set1.has(num)) {
            result.push(num);
            set1.delete(num);
        }
    }

    return result;
}

File: easy/problems/0349-intersection-of-two-arrays/solution.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        unordered_set<int> set1(nums1.begin(), nums1.end());
        vector<int> result;

        for (int num : nums2) {
            if (set1.find(num) != set1.end()) {
                result.push_back(num);
                set1.erase(num);
            }
        }

        return result;
    }
};
